Source/CutomePlayer/YoutubeVideoPlayer.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  YoutubeVideoPlayer.swift
+//  edX
+//
+//  Created by Andrey on 9/4/18.
+//  Copyright © 2018 edX. All rights reserved.
+//
+
+class YoutubeVideoPlayer: VideoPlayer{
+
+    let playerView: YTPlayerView
+    var videoId: String
+
+    override var currentTime: TimeInterval {
```

Comment:
If we're overriding currentTime do we also need to override other methods like duration, rate, isPlaying, playableDuration?

the overridden function has this body:

`player.currentItem?.currentTime().seconds ?? 0`

Do we need the false/nil check too? (the ?? 0)

Reply:
The most of those methods are using for the player control  in order to watch how long is the video , if the (play or pause )symbol should be showed etc , but youtube video player doesn't need that because is a webview that includes the controls, that's why I am not overriding that methods, I only override currentTime because this is necessary for transcripts